politeiad/backendv2/tstorebe/plugins/pi/hooks.go

```diff
@@ -249,6 +293,35 @@ func (p *piPlugin) proposalFilesVerify(files []backend.File) error {
 		}
 	}
 
+	// Validate vote & proposal metadata requirements
+	vm, err := voteMetadataDecode(files)
+	if err != nil {
+		return err
+	}
+	// In case of an RFP ensure irrelevant proposal metadata are not provided.
+	if isRFP(vm) {
+		switch {
+		case pm.Amount != 0:
+			return backend.PluginError{
+				PluginID:     pi.PluginID,
+				ErrorCode:    uint32(pi.ErrorCodeProposalAmountInvalid),
+				ErrorContext: "RFP metadata should not include an amount",
+			}
+		case pm.StartDate != 0:
+			return backend.PluginError{
+				PluginID:     pi.PluginID,
+				ErrorCode:    uint32(pi.ErrorCodeProposalStartDateInvalid),
+				ErrorContext: "RFP metadata should not include a start date",
+			}
+		case pm.EndDate != 0:
+			return backend.PluginError{
+				PluginID:     pi.PluginID,
+				ErrorCode:    uint32(pi.ErrorCodeProposalEndDateInvalid),
+				ErrorContext: "RFP metadata should not include an end date",
+			}
```

Comment:
Doesn't this need a default case?

Reply:
RFPs don't have work start/end dates.  RFP submissions do though.